eng/pipelines/docs.yml

```diff
@@ -18,8 +18,8 @@ jobs:
         workingDirectory: $(Build.BinariesDirectory)
         displayName: Download and Extract DocFX
       - pwsh: |
-          dotnet restore "$(Build.SourcesDirectory)\eng\docgeneration\assets\docgen.csproj"
-          dotnet restore "$(Build.SourcesDirectory)\eng\service.proj"
+          dotnet restore "$(Build.SourcesDirectory)\eng\docgeneration\assets\docgen.csproj" --packages $(Build.BinariesDirectory)/packages
+          dotnet restore "$(Build.SourcesDirectory)\eng\service.proj" --packages $(Build.BinariesDirectory)/packages
```

Comment:
While this might help it wasn't the direction I was expecting you to go. Have you tried doing a dotnet publish to get the all the restored binaries into one directory instead of manually copying them? That should work for the libraries you might still need to do some manual copying of the tools given they are in a non-standard layout, but I still believe it is better to do that copying in the docgen.csproj instead of in this yml file.